tests/Nox.Integration.Tests/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Domain.TestEntityOwnedRelationshipOneOrMany.g.cs

```diff
@@ -59,38 +59,66 @@ internal abstract partial class TestEntityOwnedRelationshipOneOrManyBase : Audit
-	{
-		InternalDomainEvents.Add(new TestEntityOwnedRelationshipOneOrManyUpdated(testEntityOwnedRelationshipOneOrMany));
-	}
+    protected virtual void InternalRaiseUpdateEvent(TestEntityOwnedRelationshipOneOrMany testEntityOwnedRelationshipOneOrMany)
+    {
+        InternalDomainEvents.Add(new TestEntityOwnedRelationshipOneOrManyUpdated(testEntityOwnedRelationshipOneOrMany));
+    }
 	
-	protected virtual void InternalRaiseDeleteEvent(TestEntityOwnedRelationshipOneOrMany testEntityOwnedRelationshipOneOrMany)
-	{
-		InternalDomainEvents.Add(new TestEntityOwnedRelationshipOneOrManyDeleted(testEntityOwnedRelationshipOneOrMany));
-	}
-	/// <summary>
-	/// Clears all domain events associated with the entity.
-	/// </summary>
+    protected virtual void InternalRaiseDeleteEvent(TestEntityOwnedRelationshipOneOrMany testEntityOwnedRelationshipOneOrMany)
+    {
+        InternalDomainEvents.Add(new TestEntityOwnedRelationshipOneOrManyDeleted(testEntityOwnedRelationshipOneOrMany));
+    }
+    /// <summary>
+    /// Clears all domain events associated with the entity.
+    /// </summary>
     public virtual void ClearDomainEvents()
-	{
-		InternalDomainEvents.Clear();
-	}
+    {
+        InternalDomainEvents.Clear();
+    }
 
     /// <summary>
     /// TestEntityOwnedRelationshipOneOrMany Test entity relationship to SecondTestEntityOwnedRelationshipOneOrMany OneOrMany SecondTestEntityOwnedRelationshipOneOrManies
     /// </summary>
-    public virtual List<SecondTestEntityOwnedRelationshipOneOrMany> SecondTestEntityOwnedRelationshipOneOrMany { get; set; } = new();
+    public virtual List<SecondTestEntityOwnedRelationshipOneOrMany> SecondTestEntityOwnedRelationshipOneOrMany { get; private set; } = new();
+    
+    /// <summary>
+    /// Creates a new SecondTestEntityOwnedRelationshipOneOrMany entity.
+    /// </summary>
+    public virtual void CreateRefToSecondTestEntityOwnedRelationshipOneOrMany(SecondTestEntityOwnedRelationshipOneOrMany relatedSecondTestEntityOwnedRelationshipOneOrMany)
+    {
+        SecondTestEntityOwnedRelationshipOneOrMany.Add(relatedSecondTestEntityOwnedRelationshipOneOrMany);
+    }
+    
+    /// <summary>
+    /// Deletes owned SecondTestEntityOwnedRelationshipOneOrMany entity.
+    /// </summary>
+    public virtual void DeleteRefToSecondTestEntityOwnedRelationshipOneOrMany(SecondTestEntityOwnedRelationshipOneOrMany relatedSecondTestEntityOwnedRelationshipOneOrMany)
+    {
+        if(SecondTestEntityOwnedRelationshipOneOrMany.Count() < 2)
+            throw new Exception($"The relationship cannot be deleted.");
+        SecondTestEntityOwnedRelationshipOneOrMany.Remove(relatedSecondTestEntityOwnedRelationshipOneOrMany);
+    }
+    
+    /// <summary>
+    /// Deletes all owned SecondTestEntityOwnedRelationshipOneOrMany entities.
+    /// </summary>
+    public virtual void DeleteAllRefToSecondTestEntityOwnedRelationshipOneOrMany()
+    {
+        if(SecondTestEntityOwnedRelationshipOneOrMany.Count() < 2)
```

Comment:
Could you please pay extra attention to this part. I got it from Relationship implementation. However if relationship is one or many, delete all ref  should directly throw exception instead of checking count. What do you think? 
If you agree with me I can update both owned entity and relationship delete all ref implementations. 
@ggrgovski @dejobratic @rochar

Reply:
I'm ok with this for now... until we have some real cases

 I have seen that even EF on the database level its does not enforce it, because, I Guess in many real cases although is mandatory in the initial state is not possible to have it.... for example (just guessing) I may create a workplace (physical one) in my domain, but until is ready for usage I cannot associate a Tenant although is mandatory to have one!

Reply:
@msadikkoseaupa  I would suggest to create a specific exception (avoid using System.Exception) and implement the INoxHttpException, so you can define the status code that this case should raise...